Make GrapheneFilterSetMixin compatible with django-filter 2

jayhale opened the pull request:
To support `Django >=2` `graphene-django` must support `django-filter >=2`.

Changes made:
- Revised `graphene_django.filter.GlobalIDFilter` to handle cases when the filter value is `None`.
- Removed `filter_for_reverse_field()` from `graphene_django.filter.GrapheneFilterSetMixin`
- Added back `filter_for_reverse_field()` to `graphene_django.filter.GrapheneFilterSetMixin` only for cases when `django-filter <2`
- Updated requirements in `setup.py` to `Django >=1.11` and `django-filter >=2` for `python ==3` and `django-filter >=1` for `python ==2`
- Removed unnecessary checks for `Django <1.9` from graphene_django/tests/test_converter.py
- Removed separate check for `JSONField` in compat.py
- Updated requirements in `examples/cookbook`
- Updated links to `django-filter` documentation in `docs/filtering.rst`

All tests pass.

Resolves: #484
Resolves: #464 

Related: #491

phalt commented:
👍 if this brings compatability with django 2.1!

synasius commented:
@jayhale PR looks great! Just a couple of notes: 

* django 1.11 is still supporting python 2 so you should not remove python 2.7 from the test matrix. Instead you should add to the matrix django 1.11 and python 2 and 3. See https://docs.djangoproject.com/en/2.0/releases/1.11/#python-compatibility
* Since you removed support for django < 1.11 (which is good :+1: ) you should clean the `graphene_django/compat.py` module. See here for more details https://coveralls.io/builds/18371679/source?filename=graphene_django%2Fcompat.py#L16

jayhale commented:
@synasius thanks.  I've:

- Added back a test case for Python 2.7 and Django 1.11
- Added back linting for Python 2.7 (rules appear to hit differently wether I'm on 2.7 or 3.x, so including both)
- Removed unnecessary checks for Django < 1.9 from `graphene_django/tests/test_converter.py`
- Removed separate check for `JSONField` in `compat.py`
- Fixed a typo in a link I updated in the filtering docs (all correct now)

Hope that closes this one out.  Cheers!

b99andla commented:
Waiting for a release on this one since its blocking our development! Thanks in advance!

ValentinCreative commented:
@b99andla I was in the same situation. 
I temporarily put this in my requirements.txt 
`git+https://github.com/jayhale/graphene-django.git@django-filter-2#egg=graphene-django` 
Instead of `graphene-django`
So I'm able to continue developments, if it can help :)

aldarund commented:
This should go live, its blocking use of django 2.1

aldarund commented:
@syrusakbary any plans on merging this anytime soon..?

patrick91 commented:
@jayhale can you resolve the conflicts in `docs/filtering.rst`?

jayhale commented:
Merge conflict resolved.

patrick91 commented:
Pinging @syrusakbary since I can't merge :)

aldarund commented:
Would be good to also release a new version after merge :)

syrusakbary commented:
The PR looks great! Merging

jckw commented:
Nice! When can we expect a release to PyPI @syrusakbary ?

syrusakbary commented:
By EOD today or tomorrow :)

syrusakbary commented:
A new version of Graphene Django: `2.2.0` is now on PyPI.

Here are the release notes: https://github.com/graphql-python/graphene-django/releases/tag/v2.2.0
